include/forcing/GenericDataProvider.hpp

```diff
@@ -1,17 +1,33 @@
 #ifndef NGEN_GENRIC_DATA_PROVIDER
 #define NGEN_GENRIC_DATA_PROVIDER
+#include <NGenConfig.h>
 
 #include "DataProvider.hpp"
 #include "DataProviderSelectors.hpp"
+#include <memory>
+#if NGEN_WITH_NETCDF
+  #include <netcdf>
+#endif
+
+namespace netCDF {
+    class NcFile;
+}
 
 namespace data_access
 {
     class GenericDataProvider : public DataProvider<double, CatchmentAggrDataSelector>
     {
         public:
 
+#if NGEN_WITH_NETCDF
+        /** Retrun the shared_ptr to the NcFile */
+        virtual std::shared_ptr<netCDF::NcFile> get_nc_file() {
+            return nullptr;
+        }
+#endif
+
```

Comment:
```suggestion
```
Gonna put an empty implementation of `virtual void finalize()`  in base `DataProvider`, because Justin and I turn `GenericDataProvider` into a type alias in the Forcings Engine PR